Shared/Utils/AsyncOperation.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  AsyncOperation.swift
+//  InternxtDesktop
+//
+//  Created by Robert Garcia on 27/5/24.
+//
+
+import Foundation
+class AsyncOperation: Operation {
```

Comment:
@patricioxavier8  `OperationQueues` only allows synchronous work, in order to keep using our async/await syntax, I created a new `AsyncOperation`, which will lock the queue until the async operation is performed, this can be reused in other `OperationQueue` contexts such a download queue